hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -202,6 +202,13 @@ private[sql] object SchemaConverters {
           st.foreach { f =>
             val fieldAvroType =
               toAvroType(f.dataType, f.nullable, f.name, childNameSpace)
+            val fieldBuilder = fieldsAssembler.name(f.name).`type`(fieldAvroType)
```

Comment:
This code is actually borrowed from Spark, and we try to avoid any changes to such code to make sure we're not diverging from Spark

Reply:
![image](https://user-images.githubusercontent.com/20527912/218361319-2783b730-ddea-4d7b-b7d4-ec225014e531.png)
When `extractPartitionValuesFromPartitionPath` is turned on, the StructType schema and AvroSchema differs. convertToAvroSchema is missing the default value when the field is nullable, making the table not queryable.

Reply:
From what i understand so far the issue is not in the conversion, but in the fact that we're not handling schema evolution properly in `HoodieAvroDataBlock` -- whenever we decode a record from an existing data block we should make sure that any nullable field has actually null as default value so that Avro reader is able to decode the data in case this particular field is not present

Reply:
cc @xiarixiaoyao

Reply:
I agree with @alexeykudinkin ，  we should not change the code of SchemaConverters.scala, this is the bug of logscanner